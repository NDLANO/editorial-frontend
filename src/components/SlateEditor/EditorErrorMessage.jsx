import React from 'react';
import PropTypes from 'prop-types';
import BEMHelper from 'react-bem-helper';

const classes = new BEMHelper({
  name: 'editor',
  prefix: 'c-',
});

const EditorErrorMessage = ({ msg, attributes, children }) =>
  <div>
    <div {...classes('error')} {...attributes}>
      <span>
        {msg}
      </span>
      {children}
    </div>
  </div>;

EditorErrorMessage.propTypes = {
  attributes: PropTypes.shape({
    'data-key': PropTypes.string.isRequired,
  }),
  msg: PropTypes.string.isRequired,
};

export default EditorErrorMessage;
