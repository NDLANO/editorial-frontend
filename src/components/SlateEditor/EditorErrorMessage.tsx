/**
 * Copyright (c) 2018-present, NDLA.
 *
 * This source code is licensed under the GPLv3 license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import { HTMLAttributes, MouseEventHandler, ReactNode } from 'react';
import { useTranslation } from 'react-i18next';
import styled from '@emotion/styled';
import { colors } from '@ndla/core';
import { EditorDeleteButton } from './common/EditorDeleteButton';

const StyledEditorErrorMessage = styled.div`
  position: relative;
  border: 1px solid ${colors.support.red};
  color: ${colors.support.red};
  padding: 1rem;
`;

interface Props {
  msg: string;
  onRemoveClick?: MouseEventHandler;
  attributes?: HTMLAttributes<HTMLDivElement>;
  children?: ReactNode;
}

const EditorErrorMessage = ({ msg, attributes, onRemoveClick, children }: Props) => {
  const { t } = useTranslation();

  return (
    <StyledEditorErrorMessage {...attributes} contentEditable={false}>
      {onRemoveClick && (
        <EditorDeleteButton aria-label={t('form.remove')} onClick={onRemoveClick} />
      )}
      <span>{msg}</span>
      {children}
    </StyledEditorErrorMessage>
  );
};

export default EditorErrorMessage;
