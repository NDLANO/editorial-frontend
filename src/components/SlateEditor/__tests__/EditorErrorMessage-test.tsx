/**
 * Copyright (c) 2017-present, NDLA.
 *
 * This source code is licensed under the GPLv3 license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import { render } from '@testing-library/react';
import EditorErrorMessage from '../EditorErrorMessage';
import IntlWrapper from '../../../util/__tests__/IntlWrapper';

test('EditorErrorMessage renders', () => {
  const embed = {
    message: 'This is error',
    resource: 'error',
  };

  const { container } = render(
    <IntlWrapper>
      <EditorErrorMessage msg={embed.message} onRemoveClick={() => {}} />
    </IntlWrapper>,
  );

  expect(container).toMatchSnapshot();
});
