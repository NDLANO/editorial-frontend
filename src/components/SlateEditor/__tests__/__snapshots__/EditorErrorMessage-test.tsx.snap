// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`EditorErrorMessage renders 1`] = `
.emotion-0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  border: 1px solid #d1372e;
  color: #d1372e;
  padding: 1rem;
}

.emotion-2 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  gap: 6px;
  color: #fff;
  background: #d1372e;
  border: 2px solid #d1372e;
  border-radius: 4px;
  outline-width: 0;
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: 'Source Sans Pro',Helvetica,Arial,STKaiti,'华文楷体',KaiTi,SimKai,'楷体',KaiU,DFKai-SB,'標楷體',SongTi,'宋体',sans-serif;
  font-weight: 600;
  -webkit-transition: all .2s cubic-bezier(0.17, 0.04, 0.03, 0.94);
  transition: all .2s cubic-bezier(0.17, 0.04, 0.03, 0.94);
  box-shadow: none;
  text-align: center;
  padding: 4px 6px;
  font-size: 14px;
  font-size: 0.7777777777777778rem;
  line-height: 18px;
  min-height: 32px;
  border-width: 1px;
  outline-width: 2px;
  color: #d1372e;
  background: transparent;
  border-color: transparent;
  border-radius: 100%;
  padding: 6px;
  line-height: 1;
  border-color: transparent;
  min-height: unset;
}

.emotion-2:hover,
.emotion-2:focus-visible {
  color: #fff;
  background-color: #d1372e;
  border-color: #d1372e;
}

.emotion-2[disabled] {
  color: #777777;
  background-color: #e6e6e6;
  border-color: transparent;
  cursor: not-allowed;
}

.emotion-2:where([lang='zh'], .emotion-2[lang='zh-Hans'], [lang='zh-Hant']):not([data-pinyin]) {
  font-size: calc(14px * 1.11);
  font-size: calc(0.7777777777777778rem * 1.11);
}

.emotion-2:hover,
.emotion-2:active,
.emotion-2:focus-visible {
  color: #fff;
  background: #d1372e;
  border-color: #d1372e;
}

.emotion-2[disabled] {
  color: #777777;
  background-color: transparent;
  border-color: transparent;
  cursor: not-allowed;
}

.emotion-2 svg {
  width: 24px;
  height: 24px;
  margin: 0;
}

.emotion-3 {
  display: inline-block;
  fill: currentColor;
  vertical-align: middle;
  line-height: 1em;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 16px;
  height: 16px;
}

<div>
  <div
    class="emotion-0 emotion-1"
    contenteditable="false"
  >
    <div>
      <span>
        This is error
      </span>
    </div>
    <button
      aria-label="Fjern"
      class="emotion-2"
      contenteditable="false"
      data-state="closed"
      data-testid="close-related-button"
      data-trigger="true"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="emotion-3 emotion-4"
        data-icon=""
        data-license="Apache License 2.0"
        data-source="Material Design"
        fill="currentColor"
        preserveAspectRatio="xMidYMid meet"
        viewBox="0 0 24 24"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M6 19c0 1.1.9 2 2 2h8a2 2 0 0 0 2-2V7H6zm2.5-7.1 1.4-1.4 2.1 2 2.1-2 1.4 1.4-2 2.1 2 2.1-1.4 1.4-2.1-2-2.1 2-1.4-1.4 2-2.1zm7-7.9-1-1h-5l-1 1H5v2h14V4z"
        />
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
      </svg>
    </button>
  </div>
</div>
`;
